File: backend/src/main/java/backend/task/TaskDto.java
package backend.task;

import java.time.LocalDateTime;
import java.util.List;

import lombok.Data;

@Data
public class TaskDto {
    private Long id;
    private String name;
    private String description;
    private Integer points;
    private Boolean checked;
    private LocalDateTime timestamp;
    private String repeatEvery;
    private LocalDateTime repeatUntil;
    private Long imageId;
    private List<Long> assignedUserIds;
    private List<String> assignedUserNames;
}

File: frontend/src/app/features/task/models/TaskDto.ts
export interface TaskDTO {
  id: number;
  name: string;
  description: string;
  points: number;
  checked: boolean;
  
  // Dates are received as strings (ISO 8601) from JSON
  timestamp: string; 
  repeatUntil: string;
  
  repeatEvery: 'Daily' | 'Weekly' | 'Monthly' | string;

  // Foreign Key references (IDs) instead of full objects
  imageId?: number; 
}